kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/Constraints.java

```diff
@@ -456,6 +456,21 @@ public String toString() {
     return query;
   }
 
+  /**
+   * Get a normalized query map for the constraints. A normalized query map will
+   * be equal in value and iteration order for any logically equivalent set of
+   * constraints.
+   */
+  public Map<String, String> toNormalizedQueryMap() {
```

Comment:
Minor: this could be a bit simpler if you used a private implementation method and passed in the map container to use. Then toQueryMap could pass in a LinkedHashMap and the normalized version could use a TreeMap.